### PR TITLE
Add CSW harvest test

### PR DESCRIPTION
**Pull request by jbrown-xentity**
Add CSW harvest test with public endpoint.
Document for future issues.
Update wait timeout for 10 minutes,
as the csw harvest takes longer to complete.

Related to https://github.com/GSA/datagov-deploy/issues/3376

### CODE REVIEW

File: e2e/cypress/integration/000_harvest.spec.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ describe('Harvest', () => {
     const harvestOrg = 'test-harvest-org'
     const dataJsonHarvestSoureName = 'test-harvest-datajson'
     const wafIsoHarvestSourceName = 'test-harvest-waf-iso'
+    const cswHarvestSoureName = 'test-harvest-csw'
```
Comment (mogul) [inline e2e/cypress/integration/000_harvest.spec.js:6]:
```suggestion
    const cswHarvestSourceName = 'test-harvest-csw'
```

File: e2e/cypress/integration/zzz_cleanup.spec.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ describe('Cleanup site', () => {
     const harvestOrg = 'test-harvest-org'
     const dataJsonHarvestSoureName = 'test-harvest-datajson'
     const wafIsoHarvestSourceName = 'test-harvest-waf-iso'
+    const cswHarvestSoureName = 'test-harvest-csw'
```
Comment (mogul) [inline e2e/cypress/integration/zzz_cleanup.spec.js:5]:
```suggestion
    const cswHarvestSourceName = 'test-harvest-csw'
```

File: e2e/cypress/integration/zzz_cleanup.spec.js [hunk 3]
```diff
@@ -12,6 +13,7 @@ describe('Cleanup site', () => {
         // Clear and remove all harvested data
         cy.delete_harvest_source(dataJsonHarvestSoureName);
         cy.delete_harvest_source(wafIsoHarvestSourceName);
+        cy.delete_harvest_source(cswHarvestSoureName);
```
Comment (mogul) [inline e2e/cypress/integration/zzz_cleanup.spec.js:16]:
```suggestion
        cy.delete_harvest_source(cswHarvestSourceName);
```

**Review by mogul — CHANGES_REQUESTED** [review]

File: e2e/cypress/integration/000_harvest.spec.js [hunk 4]
```diff
@@ -110,4 +111,37 @@ describe('Harvest', () => {
...
+         * how many harvest sources are created for CSW by going to
+         * https://catalog.data.gov/harvest?source_type=csw
+         */
+        cy.visit(`/harvest/new`);
+        
+        // Hide flask debug toolbar
+        cy.get('#flHideToolBarButton').click();
+
+        cy.create_harvest_source('https://portal.opentopography.org/geoportal/csw',
+                        cswHarvestSoureName,
```
Comment (nickumia-reisys) [inline e2e/cypress/integration/000_harvest.spec.js:134]:
```suggestion
                        cswHarvestSourceName,
```

File: e2e/cypress/integration/000_harvest.spec.js [hunk 5]
```diff
@@ -110,4 +111,37 @@ describe('Harvest', () => {
...
+                        'cypress test csw',
+                        'csw',
+                        'False',
+                        false)
+
+        // harvestTitle must not contain spaces, otherwise the URL redirect will not confirm
+        cy.location('pathname').should('eq', '/harvest/' + cswHarvestSoureName)
```
Comment (nickumia-reisys) [inline e2e/cypress/integration/000_harvest.spec.js:141]:
```suggestion
        cy.location('pathname').should('eq', '/harvest/' + cswHarvestSourceName)
```

File: e2e/cypress/integration/000_harvest.spec.js [hunk 6]
```diff
@@ -110,4 +111,37 @@ describe('Harvest', () => {
...
+    })
+
+    it('Start CSW Harvest Job', () => {
+        cy.start_harvest_job(cswHarvestSoureName)
```
Comment (nickumia-reisys) [inline e2e/cypress/integration/000_harvest.spec.js:145]:
```suggestion
        cy.start_harvest_job(cswHarvestSourceName)
```

**Review by nickumia-reisys — CHANGES_REQUESTED** [review]
Found the other references that @mogul missed 👁️

**Comment by FuhuXia** [comment]
Any alternative to adding the external dependency on opentopography.org CSW endpoint?

**Review by nickumia-reisys — APPROVED** [review]

**Review by mogul — APPROVED** [review]

**Pull request merged**